kernel/Kconfig

```diff
@@ -780,6 +780,7 @@ config USE_SWITCH_SUPPORTED
 config SMP
 	bool "Enable symmetric multithreading support"
 	depends on USE_SWITCH
+	depends on (ARM_PSCI && ARMV8_A) || !ARMV8_A
```

Comment:
I think now the dependency can be decoupled. ie: one can still have SMP on V8 but no PSCI.